lib/autoupdate/tools/updater.go

```diff
@@ -203,6 +204,15 @@ func (u *Updater) CheckRemote(ctx context.Context, proxyAddr string, insecure bo
 		return "", false, trace.Wrap(err)
 	}
 
+	m := modules.GetModules()
+	envBaseURL := os.Getenv(autoupdate.BaseURLEnvVar)
+	if resp.Edition == modules.BuildOSS && m.BuildType() == modules.BuildCommunity && envBaseURL == "" {
```

Comment:
```suggestion
	if m.BuildType() == modules.BuildOSS && envBaseURL == "" {
```
I'm not sure I understand the logic implemented here. If the cluster is AGPL and tsh is community, it seems like it's already too late -- the user is running licensed code. 

I think we just want to prevent an AGPL build from updating to a community build.

Reply:
Oh, apologies, I see there's already a check for this case below.

I don't think we need to check the server edition. Given that tsh could be used on multiple clusters at the same time, it seems sufficient to ensure that the edition that was originally installed does not change.